internal/services/resource/resource_provider_registration_resource.go

```diff
@@ -304,7 +305,28 @@ func (r ResourceProviderRegistrationResource) Delete() sdk.ResourceFunc {
 				return err
 			}
 
-			err = r.applyFeatures(ctx, metadata, *id, metadata.ResourceData.Get("feature").(*pluginsdk.Set).List(), make([]interface{}, 0))
+			existingFeatures := make([]interface{}, 0)
+			result, err := featureClient.ListComplete(ctx, id.ProviderName)
+			if err != nil {
+				return fmt.Errorf("retrieving features for %s: %+v", *id, err)
+			}
+			for result.NotDone() {
+				value := result.Value()
+				if value.Properties != nil && value.Properties.State != nil && value.Name != nil {
+					featureName := (*value.Name)[len(id.ProviderName)+1:]
+					switch *value.Properties.State {
+					case Registering, Registered:
+						existingFeatures = append(existingFeatures, map[string]interface{}{"name": featureName, "registered": true})
+					case Unregistering, Unregistered:
+						existingFeatures = append(existingFeatures, map[string]interface{}{"name": featureName, "registered": false})
+					}
+				}
+				if err := result.NextWithContext(ctx); err != nil {
+					return fmt.Errorf("enumerating features: %+v", err)
+				}
+			}
+
+			err = r.applyFeatures(ctx, metadata, *id, existingFeatures, make([]interface{}, 0))
```

Comment:
#23072 ports this over to using `hashicorp/go-azure-sdk` - can we revert this so this doesn't conflict?

Reply:
sure